src/gt4py/storage/storage.py

```diff
@@ -14,604 +14,1112 @@
+        alignment=alignment,
+        gpu=gpu,
+        layout_map=layout_map,
+        copy=copy,
+        device_data=device_data,
+        managed=managed,
+    )
 
-    __array_subok__ = True
 
-    def __new__(cls, shape, dtype, backend, default_origin, mask=None):
-        """"
+class Storage(abc.ABC, np.lib.mixins.NDArrayOperatorsMixin):
+
+    SUPPORTED_UFUNCS = {
+        # math operations
+        np.add,
+        np.subtract,
+        np.multiply,
+        np.divide,
+        np.logaddexp,
+        np.logaddexp2,
+        np.true_divide,
+        np.floor_divide,
+        np.negative,
+        np.positive,
+        np.power,
+        np.remainder,
+        np.mod,
+        np.fmod,
+        np.divmod,
+        np.absolute,
+        np.fabs,
+        np.rint,
+        np.sign,
+        np.heaviside,
+        np.conj,  # note: complex types not supported
+        np.conjugate,  # note: complex types not supported
+        np.exp,
+        np.exp2,
+        np.log,
+        np.log2,
+        np.log10,
+        np.expm1,
+        np.log1p,
+        np.sqrt,
+        np.square,
+        np.cbrt,
+        np.reciprocal,
+        np.gcd,
+        np.lcm,
+        # trigonometric functions
+        np.sin,
+        np.cos,
+        np.tan,
+        np.arcsin,
+        np.arccos,
+        np.arctan,
+        np.arctan2,
+        np.hypot,
+        np.sinh,
+        np.cosh,
+        np.tanh,
+        np.arcsinh,
+        np.arccosh,
+        np.arctanh,
+        np.deg2rad,
+        np.rad2deg,
+        # bit-twiddling functions
+        np.bitwise_and,
+        np.bitwise_or,
+        np.bitwise_xor,
+        np.invert,
+        np.left_shift,
+        np.right_shift,
+        # comparison functions
+        np.greater,
+        np.greater_equal,
+        np.less,
+        np.less_equal,
+        np.not_equal,
+        np.equal,
+        np.logical_and,
+        np.logical_or,
+        np.logical_xor,
+        np.logical_not,
+        np.maximum,
+        np.minimum,
+        np.fmax,
+        np.fmin,
+        # floating functions
+        np.isfinite,
+        np.isinf,
+        np.isnan,
+        # np.isnat,  # times not supported
+        np.fabs,
+        np.signbit,
+        np.copysign,
+        np.nextafter,
+        np.spacing,
+        np.modf,
+        np.ldexp,
+        np.frexp,
+        np.fmod,
+        np.floor,
+        np.ceil,
+        np.trunc,
+    }
+
+    def __new__(
+        cls,
+        data,
+        *,
+        default_parameters=None,
+        default_origin=None,
+        shape=None,
+        dtype=None,
+        axes=None,
+        alignment=None,
+        gpu=None,
+        layout_map=None,
+        copy=True,
+        device_data=None,
+        managed=None,
+    ):
+        # 1) for each parameter assert that type and value is valid
+        if data is not None:
+            if not isinstance(data, Storage):
+                try:
+                    np.asarray(data)
+                except:
```

Comment:
It can be dangerous to use blanket except clauses. I would strongly recommend restricting this to the expected exception type. You never know what kind of exceptions might occur that will either silently be ignored or be hidden as a different exception when using blanket `except:`.